pkg/fieldpath/object/object.go

```diff
@@ -0,0 +1,94 @@
+/*
+Copyright 2021 The Crossplane Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package object
+
+import (
+	"k8s.io/apimachinery/pkg/runtime"
+
+	v1 "github.com/crossplane/crossplane-runtime/apis/common/v1"
+	"github.com/crossplane/crossplane-runtime/pkg/fieldpath"
+)
+
+// ToPaved tries to convert a runtime.Object into a *Paved via
+// runtime.DefaultUnstructuredConverter if needed. Returns the paved if
+// the conversion is successful along with whether the
+// runtime.DefaultUnstructuredConverter has been employed during the
+// conversion.
```

Comment:
I see that when you call this function you name the `bool` return value `copied`. It would be useful to include this information in the docstring. From my reader's perspective it is not obvious what the implications of `runtime.DefaultUnstructuredConverter` being employed is - I can only infer from the function below that it means the returned value is a copy when it would otherwise not be.

Reply:
Also, note that there is already a `fieldpath.PaveObject` function that has very similar functionality to this one. The key difference being that it always invokes the `DefaultUnstructuredConverter`. Do we need both?

Reply:
If we decide to keep `ToPaved`, I will update the documentation as you suggested above.

Reply:
Thanks! I don't have a strong opinion either. Usually I would bias for API simplicity over performance optimisations though, at least until we experience noticeable performance issues. In this case I'm happy for you to choose - we can either deprecate the old function or not add the new one.

Reply:
Removed `fieldpath.ToPaved` and kept `fieldpath.PaveObject`.